mxm_wifiex/wlan_src/mlinux/moal_shim.c

```diff
@@ -1033,13 +1033,19 @@ mlan_status moal_ioctl_complete(t_void *pmoal, pmlan_ioctl_req pioctl_req,
 		return MLAN_STATUS_SUCCESS;
 	}
 
-	if (status != MLAN_STATUS_SUCCESS && status != MLAN_STATUS_COMPLETE)
-		PRINTM(MERROR,
-		       "IOCTL failed: %p id=0x%x, sub_id=0x%x action=%d, status_code=0x%x\n",
-		       pioctl_req, pioctl_req->req_id,
-		       (*(t_u32 *)pioctl_req->pbuf), (int)pioctl_req->action,
-		       pioctl_req->status_code);
-	else
+	if (status != MLAN_STATUS_SUCCESS && status != MLAN_STATUS_COMPLETE) {
+		if (pioctl_req->status_code == MLAN_ERROR_CMD_CANCEL)
+			PRINTM(MIOCTL,
+						 "IOCTL canceled: %p id=0x%x, sub_id=0x%x action=%d",
+						 pioctl_req, pioctl_req->req_id,
+						 (*(t_u32 *)pioctl_req->pbuf), (int)pioctl_req->action);
+		else
+			PRINTM(MERROR,
+						 "IOCTL failed: %p id=0x%x, sub_id=0x%x action=%d, status_code=0x%x\n",
+						 pioctl_req, pioctl_req->req_id,
+						 (*(t_u32 *)pioctl_req->pbuf), (int)pioctl_req->action,
+						 pioctl_req->status_code);
```

Comment:
Perhaps something like this would be better (careful - untested!):

```suggestion
		PRINTM(MERROR,
					 "IOCTL %s: %p id=0x%x, sub_id=0x%x action=%d, status_code=0x%x\n",
					 (pioctl_req->status_code == MLAN_ERROR_CMD_CANCEL) ? "canceled" : "failed",
					 pioctl_req, pioctl_req->req_id,
					 (*(t_u32 *)pioctl_req->pbuf), (int)pioctl_req->action,
					 pioctl_req->status_code);
```

This way we always give the `status_code` which might be beneficial.

Reply:
won't be :-)
the point of the patch is the "PRINTM(MIOCTL," vs "PRINTM(MERROR,"
the added "IOCTL canceled" message is just a bonus

Reply:
> won't be :-) the point of the patch is the "PRINTM(MIOCTL," vs "PRINTM(MERROR," the added "IOCTL canceled" message is just a bonus

True, missed that part. :) I guess this whole reporting part in this driver needs some TLC, but in this particular case it's fine to solve it this way.

Reply:
thnx :-)
"tender love and care"... more like "Ceterum autem censeo mwifiex esse delendam" ;-)